netlogo-gui/src/main/app/codetab/ProceduresMenu.scala

```diff
@@ -51,23 +75,22 @@ class ProceduresMenu(target: ProceduresMenuTarget)
       }
     })
 
+    repopulate(menu, filterField, items)
+  }
+
+  private def repopulate(menu: JPopupMenu, filterField: JTextField, items: Seq[JMenuItem]): Unit = {
+    val query = filterField.getText
+    val caseSensitive = if (query == query.toLowerCase) "" else "(?i)"
+    val pattern = caseSensitive + filterField.getText.split("").mkString(".*", ".*", ".*")
+    menu.removeAll()
     menu.add(filterField)
-    if(procs.isEmpty) menu.add(new JMenuItem("<"+I18N.gui.get("tabs.code.procedures.none")+">") { setEnabled(false) })
-    else {
-      for(proc <- procs.sortWith(_.toUpperCase < _.toUpperCase)) {
-        val item = new JMenuItem(proc)
-        val namePos = procsTable(proc).identifier.start
-        val end  = procsTable(proc).endKeyword.end
-        item.addActionListener{() =>
-          // invokeLater for the scrolling behavior we want. we scroll twice: first bring the end into
-          // view, then bring the beginning into view, so then we can see both, if they fit - ST 11/4/04
-          target.select(end, end)
-          EventQueue.invokeLater{() =>
-            target.select(namePos, namePos + proc.size)  // highlight the name
-          }
-        }
-        menu.add(item)
-      }
-    }
+    items.filter(_.getText matches pattern).foreach(menu.add)
```

Comment:
Lines 89-93 take care of adding the items, don't they? Won't this line cause the items to be added twice?